deploy/eck-integrations/templates/eck-agent/elastic-agent-daemonset-secret.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: v1
+kind: Secret
```

Comment:
Why is this a secret and not configmap? Just for my understanding

Reply:
Yeah it'd be best if we could use both a configmap for most of the config and reference the secrets for the ES connection details using Agent's k8s secret provider.

Reply:
this is a secret guys because we use `configRef` from ECK operator and this associates the config from a secret with `secretName` property. We could switch to using `config` field of the CRD but this would require embedding the config yaml inside the agent CRD and it imo this would be overloaded. On the plus side having ElasticSearch creds/api_key saved in a secret file is more aligned with k8s practices

Reply:
Fine by me